messaging/react-typescript-webpack/src/service-worker/firebase-messaging-sw.ts

```diff
@@ -0,0 +1,72 @@
+// Tell TypeScript that we are in Service Worker context, which is a specific
+// type of Web Worker that has extra APIs available.
+// https://github.com/Microsoft/TypeScript/issues/14877
+declare var self: ServiceWorkerGlobalScope;
+
+import firebase from 'firebase/app';
+import 'firebase/messaging';
+import { firebaseConfig } from '../common/configuration';
+
+firebase.initializeApp(firebaseConfig);
+
+const messaging = firebase.messaging();
```

Comment:
any reason not to use the same getMessaging() helper than the window version does?

Reply:
VAPID key is only set in the window. But I should still check if messaging is supported. Added that.